### ISSUE TITLE
scoop mpv location

### ISSUE THREAD

**Issue by SynapseScribe** [pierretom/autoselect-forced-sub#1]
where should i install lua script for scoop version of mpv

installation folder is:
C:\Users\<User>\scoop\apps\mpv-git\current

i tried copying the lua script in C:\Users\<User>\AppData\Roaming\mpv\scripts\
but it's not taking effect

**Comment by pierretom** [comment]
You should be able to place it in the installation folder, try in `C:\Users<User>\scoop\apps\mpv-git\current\mpv\scripts` or `C:\Users<User>\scoop\apps\mpv-git\current\scripts`.
Otherwise there is the --script option too: `mpv --script=path_to_script.lua`.

**Comment by SynapseScribe** [comment]
i put it in both places but is not taking effect

**Comment by pierretom** [comment]
Can you show the output of:
`mpv.exe --version`
and
`mpv.exe --script=path\to\autoselect-forced-sub.lua`

**Comment by pierretom** [comment]
https://www.youtube.com/watch?v=RnPgI3E1mT8
In this video he places the scripts in scoop\apps\mpv\current\portable_config\scripts

Also in the manpage:

> If a directory named portable_config next to the mpv.exe exists, all config will be loaded from this directory only. Watch later config files and cache files are written to this directory as well. (This exists on Windows only and is redundant with $MPV_HOME. However, since Windows is very scripting unfriendly, a  wrapper  script  just  setting $MPV_HOME, like you could do it on other systems, won't work. portable_config is provided for convenience to get around this restriction.)

**Comment by SynapseScribe** [comment]
I copied the 2 files under:

[2025-05-30 20:45:15] PS> ls

    Directory: C:\Users\<User>\scoop\apps\mpv-git\current\portable_config

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----           10-Apr-25  1:20 AM                cache
d----           30-May-25 19:01 PM                scripts
-a---           30-May-25 19:01 PM          15858 autoselect-forced-sub.lua

[2025-05-30 20:45:17] PS> cd .\scripts\
[2025-05-30 20:45:18] PS> ls

    Directory: C:\Users\<User>\scoop\apps\mpv-git\current\portable_config\scripts

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           30-May-25 19:01 PM            127 afs.conf




 mpv --version
mpv v0.40.0-119-g1d1535ff9 Copyright © 2000-2025 mpv/MPlayer/mplayer2 projects
 built on May 23 2025 00:13:58
libplacebo version: v7.351.0 (v7.351.0-dirty)
FFmpeg version: N-119684-g670089304
FFmpeg library versions:
   libavcodec      62.3.101
   libavdevice     62.0.100
   libavfilter     11.0.100
   libavformat     62.0.102
   libavutil       60.3.100
   libswresample   6.0.100
   libswscale      9.0.100




[2025-05-30 19:16:24] PS> mpv.exe --script=C:\Users\<User>\scoop\apps\mpv-git\20250525\portable_config\autoselect-forced-sub.lua
[2025-05-30 19:53:03] PS>

the second command shows no output but starts the player.

When I load an .mkv in the player, the eng subtitle is not loaded

**Comment by pierretom** [comment]
Place **autoselect-forced-sub.lua** in the folder C:\Users\<User>\scoop\apps\mpv-git\current\portable_config\scripts
Place **afs.conf** (optional) in the folder C:\Users\<User>\scoop\apps\mpv-git\current\portable_config\script-opts

**Comment by SynapseScribe** [comment]
i did so and still when i open the mkv, the subtitle is not auto selected. these are the subtitles embedded in the mkv

![Image](https://github.com/user-attachments/assets/3d270d50-b250-46d7-a774-290627c06314)

![Image](https://github.com/user-attachments/assets/71ecea0c-6a2a-4fdd-bc17-e7ffdbebf864)

also tried the  mpv.exe --script ... method

**Comment by pierretom** [comment]
Show the output, I need to see what happens when you load the media (hide the media name if you prefer).

**Comment by SynapseScribe** [comment]
the output of what? when i load the media it looks like above (it shows subtitle -/2, meaning it was not loaded)

**Comment by SynapseScribe** [comment]
if you mean the output of command, it doesn't output anything. command ends when player opens and before i load the media (drag/drop)

[2025-05-30 21:36:21] PS> mpv.exe --script=C:\Users\<user>\scoop\apps\mpv-git\20250525\portable_config\scripts\autoselect-forced-sub.lua
[2025-05-30 22:03:46] PS>

**Comment by pierretom** [comment]
Yes the output of the command, e.g. when you start mpv from PowerShell.
You can also try: `mpv.exe --log-file=mpv.log` and paste the content of the file.

**Comment by SynapseScribe** [comment]
[mpv.log](https://github.com/user-attachments/files/20525791/mpv.log)

log for:
mpv.exe --script=C:\Users\<user>\scoop\apps\mpv-git\20250525\portable_config\scripts\autoselect-forced-sub.lua --log-file=mpv.log
opening test.mkv (drag/drop)
then closing player

**Comment by pierretom** [comment]
You loaded the script twice.
Otherwise:
> [  10.528][i][autoselect_forced_sub] Not found only forced subtitles, try with --script-opts=afs-sub_forced_only=no

Your subtitle is not a forced subtitle, that's why it's not selected. To select forced and non forced subtitles use `--script-opts=afs-sub_forced_only=no` (see in the readme how to set it in the configuration files).

**Comment by SynapseScribe** [comment]
mpv.exe --script=C:\Users\user\scoop\apps\mpv-git\20250525\portable_config\scripts\autoselect-forced-sub.lua --log-file=mpv.log --script-opts=afs-sub_forced_only=no 
worked !

by forced you mean external loaded subtitles, right?

i edited afs.conf to sub_forced_only=no and now it works always, without launching from CLI

thank you !

**Comment by pierretom** [comment]
> by forced you mean external loaded subtitles, right?

No, see the definition in the readme.

> i edited afs.conf to sub_forced_only=no and now it works always, without launching from CLI

Ok, you don't need to use the `--script` option if the Lua script is installed in the scripts folder. That's probably why it's loaded twice.

**Comment by SynapseScribe** [comment]
> > by forced you mean external loaded subtitles, right?
> 
> No, see the definition in the readme.

I have read it now, but not sure i entirely understand. However, if in my case works with sub_forced_only=no, i wonder why this setting shouldn't be set as default in the afs.conf in the repo.

> 
> > i edited afs.conf to sub_forced_only=no and now it works always, without launching from CLI
> 
> Ok, you don't need to use the `--script` option if the Lua script is installed in the scripts folder. That's probably why it's loaded twice.

I think that was because the scripts were already in the right place when i ran the command to start mpv with --script. So the script was loaded twice, once from file, and once from cmd.

**Comment by pierretom** [comment]
> > > by forced you mean external loaded subtitles, right?
> > 
> > 
> > No, see the definition in the readme.
> 
> I have read it now, but not sure i entirely understand. However, if in my case works with sub_forced_only=no, i wonder why this setting shouldn't be set as default in the afs.conf in the repo.

Forced subtitles are not a full translation of the audio track. They can appear sometime in video for example to translate a visual word that is not in the same language as the audio track. If you have an English audio track and you see in a video the word "boulangerie", the forced subtitle can translate it to "bakery".
